tests/base.go

```diff
@@ -2,65 +2,56 @@ package tests
 
 import (
 	"bytes"
-	"fmt"
-	"io/ioutil"
-	"log"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
 	"net"
 	"os"
 	"os/exec"
 	"strings"
 	"time"
 
-	"github.com/goombaio/namegenerator"
+	"github.com/pkg/errors"
 	"golang.org/x/crypto/ssh"
 )
 
-// UpWg used for up wireguard
-func UpWg(wgConfig string) {
+// UpWg used for setting up the wireguard interface
+func UpWg(wgConfig string) (string, error) {
```

Comment:
parameterize the path where to touch wireguard config file, typically the caller can pass the name, and in the test you can either use TempDir or https://go.dev/src/io/ioutil/tempfile.go